pkg/helm/option.go

```diff
@@ -361,6 +368,80 @@ func WithMaxHistory(max int32) HistoryOption {
 
 // NewContext creates a versioned context.
 func NewContext() context.Context {
-	md := metadata.Pairs("x-helm-api-client", version.Version)
+	//md := metadata.Pairs("x-helm-api-client", version.Version,)
+	contextData := getMapFromKubeConfig()
+	contextData = append(contextData, "x-helm-api-client")
+	contextData = append(contextData, version.Version)
+	md := metadata.Pairs(contextData...)
 	return metadata.NewContext(context.TODO(), md)
 }
+
+func getMapFromKubeConfig() []string {
+	var contextData []string
+	flags := pflag.NewFlagSet("", pflag.ContinueOnError)
+	flags.SetNormalizeFunc(utilflag.WarnWordSepNormalizeFunc) // Warn for "_" flags
+	clientConfig := cmdutil.DefaultClientConfig(flags)
+	c, err := clientConfig.ClientConfig()
+	if err != nil {
+		log.Println(err)
+		return nil
+	}
+	// bearer token
+	if len(c.BearerToken) != 0 {
+		contextData = append(contextData, "authorization")
+		contextData = append(contextData, "Bearer "+c.BearerToken)
+	}
+	if len(c.Host) != 0 {
+		contextData = append(contextData, "k8s-server")
+		contextData = append(contextData, c.Host)
+	}
+	if len(string(c.CAData)) != 0 {
+		contextData = append(contextData, "k8s-certificate-authority")
+		contextData = append(contextData, base64.StdEncoding.EncodeToString(bytes.TrimSpace(c.CAData)))
```

Comment:
I think it is better to avoid reading KubeConfig directly. https://github.com/kubernetes/client-go/blob/master/tools/clientcmd/api/v1/types.go#L84  Things are bit complicated and seems can change in future.